autoPyTorch/api/base_task.py

```diff
@@ -1,3 +1,18 @@
+"""Base class for tasks to solve
+* The shared components among all the tasks
+* This module provides the optimization given a pipeline
+* This module plays a role of communicating with
+  distributed clients
+
```

Comment:
We can also mention that its the main API to interact with the user and fits the ensemble.